lib/backend/path_parts.dart

```diff
@@ -1,48 +1,46 @@
 class PathParts {
-  final String root;
-  final List<String> parts;
-  final String separator;
-
   const PathParts(this.root, this.parts, this.separator);
 
   factory PathParts.parse(String path) {
     late final String root;
     late final String separator;
 
-    if (path.startsWith("/")) {
-      root = "/";
-      separator = "/";
-    } else if (path.startsWith(RegExp(r"[A-Z]:\\", caseSensitive: false))) {
-      root = RegExp(r"^[A-Z]:\\", caseSensitive: false)
+    if (path.startsWith('/')) {
+      root = '/';
+      separator = '/';
+    } else if (path.startsWith(RegExp(r'[A-Z]:\\', caseSensitive: false))) {
+      root = RegExp(r'^[A-Z]:\\', caseSensitive: false)
           .firstMatch(path)!
           .group(0)!;
-      separator = "\\";
+      separator = '\\';
     }
-    final String cleanPath = path.replaceFirst(root, "");
+    final cleanPath = path.replaceFirst(root, '');
 
     late final List<String> parts;
     if (cleanPath.isNotEmpty) {
-      parts = path.replaceFirst(root, "").split(separator);
+      parts = path.replaceFirst(root, '').split(separator);
     } else {
       parts = [];
     }
 
     return PathParts(root, parts, separator);
   }
+  final String root;
+  final List<String> parts;
+  final String separator;
```

Comment:
i'm sorry but i really don't like this style of fields after constructor, especially with the factory one here. wouldn't it be better to know what the class represents before knowing how it's constructed? i'd prefer to be able to tell what the class contains at a glance

Reply:
as mentioned above, I didnt do this manually

this is a lint rule and dart fix --apply

if you dont like it, I'll find the rule and disable it

Reply:
just reverted and removed this lint rule